pkg/controller/flight/factory.go

```diff
@@ -28,28 +32,38 @@ func (f *flightReconcilerFactory) FlightReconciler(kluster *v1.Kluster) (FlightR
 		return nil, err
 	}
 
-	nodes, err := f.getNodes(kluster, client)
+	instances, err := f.getInstances(kluster, client)
+	if err != nil {
+		return nil, err
+	}
+
+	lister, err := f.NodeObservatory.GetListerForKluster(kluster)
+	if err != nil {
+		return nil, err
+	}
+
+	nodes, err := lister.List(labels.Everything())
 	if err != nil {
 		return nil, err
 	}
 
 	var reconciler FlightReconciler
-	reconciler = &flightReconciler{kluster, nodes, client, f.Logger}
+	reconciler = &flightReconciler{kluster, instances, nodes, client, f.Logger}
 	reconciler = &LoggingFlightReconciler{reconciler, f.Logger}
 	return reconciler, nil
 }
 
-func (d *flightReconcilerFactory) getNodes(kluster *v1.Kluster, client openstack_kluster.KlusterClient) ([]Node, error) {
-	nodes := []Node{}
+func (d *flightReconcilerFactory) getInstances(kluster *v1.Kluster, client openstack_kluster.KlusterClient) ([]Instance, error) {
```

Comment:
maybe we should put the getInstances for whole cluster in the client as well. and not just for single pool